lib/utils/fncache.go

```diff
@@ -95,7 +104,7 @@ func (c *FnCache) removeExpiredLocked(now time.Time) {
 // block until the first call updates the entry.  Note that the supplied context can cancel the call to Get, but will
 // not cancel loading.  The supplied loadfn should not be canceled just because the specific request happens to have
 // been canceled.
-func (c *FnCache) Get(ctx context.Context, key interface{}, loadfn func() (interface{}, error)) (interface{}, error) {
+func (c *FnCache) Get(ctx context.Context, key interface{}, loadfn func(ctx context.Context) (interface{}, error)) (interface{}, error) {
```

Comment:
If `Get` is called after the `FnCache` context is canceled, we should probably return an explicit error indicating the the cache is permanently closed, otherwise use after closure might be tricky to debug (just a lot of generic context cancellation errors).  Something to the effect of `fncache permanently closed` would be helpful.